docker/benchmark-runner/Dockerfile

```diff
@@ -40,7 +40,15 @@ FROM gcr.io/fuzzbench/builders/$fuzzer/$benchmark AS builder
 # runner.Dockerfile of each fuzzer.
 FROM gcr.io/fuzzbench/runners/$fuzzer/$benchmark-intermediate
 
+# Install runtime dependencies for benchmarks.
```

Comment:
Would it make send to do only the docker build changes (the elimination of the double dependency) in this PR, and the modification of the run_experiment/worker python code in another?
I.e. in this PR:
- This change that the comment is on.
- Delete docker/base-runner
- Update the image_types.yaml
- Modify docker/generate_makefile.py  to get the change tested with our CI. 
- Modify docker/gcb yaml files for consistency

Reply:
> do only the docker build changes (the elimination of the double dependency) in this PR, and the modification of the run_experiment/worker python code in another?

I agree to separate them. I think this PR will not kept for merging, which is merging into my branch.
I will create another one to directly merge into google/master.

My concern is, it seems https://github.com/google/fuzzbench/pull/584 will also change a lot in build logic. Then it may have conflicts.

Reply:
> > do only the docker build changes (the elimination of the double dependency) in this PR, and the modification of the run_experiment/worker python code in another?
> 
> I agree to separate them. I think this PR will not kept for merging, which is merging into my branch.
> I will create another one to directly merge into google/master.
> 
> My concern is, it seems [google#584](https://github.com/google/fuzzbench/pull/584) will also change a lot in build logic. Then it may have conflicts.

Yes, I think we should be able to land that soon.

Reply:
Here is the separation: https://github.com/google/fuzzbench/pull/601